### PR TITLE
UCP/PROTO: Add request debugging and profiling

### PR DESCRIPTION
**Pull request by yosefe**
## Why
- Implement UCP request info string for new protocols
- Add profiling for potentially time-consuming steps during a request lifetime

### CODE REVIEW

**Comment by yosefe** [comment]
@brminich can you pls take a look?

File: src/ucp/rndv/proto_rndv.c [hunk 1]
```diff
@@ -439,8 +439,6 @@ ucp_proto_rndv_send_reply(ucp_worker_h worker, ucp_request_t *req,
                   ucp_operation_names[op_id], req->send.rndv.remote_address,
                   req->send.rndv.remote_req_id,
                   req->send.proto_config->proto->name);
-
-    ucp_request_send(req);
```
Comment (brminich) [inline src/ucp/rndv/proto_rndv.c:443]: why removed?

Comment (yosefe) [reply src/ucp/rndv/proto_rndv.c:443]: it was moved outside this function, so could set req->recv.proto_rndv_config before calling `ucp_request_send(req)` (which may complete and release the request)

Comment (brminich) [reply src/ucp/rndv/proto_rndv.c:443]: ok, ic

**Review by brminich — APPROVED** [review]

**Pull request merged**